crates/anvil/core/src/eth/transaction/optimism.rs

```diff
@@ -1,11 +1,15 @@
-use alloy_network::TxKind;
-use alloy_primitives::{Address, Bytes, B256, U256};
-use alloy_rlp::{Decodable, Encodable, Error as DecodeError, Header as RlpHeader};
+use alloy_consensus::TxType;
+use alloy_network::{Transaction, TxKind};
+use alloy_primitives::{Address, Bytes, ChainId, Signature, B256, U256};
+use alloy_rlp::{
+    length_of_length, Decodable, Encodable, Error as DecodeError, Header as RlpHeader,
+};
+use std::mem;
 
 #[derive(Clone, Debug, PartialEq, Eq)]
 pub struct DepositTransactionRequest {
```

Comment:
we should move this to separate crate eventually, maybe to `alloy-rs/optimism` and have it define the tx types and impl the network / tranasction traits

Reply:
Issue for this https://github.com/foundry-rs/foundry/issues/6860